Update SamcoPreferences.cpp and OpenFIREFeedback.cpp

alessandro-satanassi opened the pull request:
x SamcoPreferences.cpp
The EEPROM.update method does not exist in other EEPROM libraries such as the one for the esp32, while the EEPROM.put method exists for all libraries. The EEPROM.update method in rp2040 does nothing but call the EEPROM.put method and nothing more. I have adapted OpenFIRE to work also on EPS32 and if there are no problems, I ask the courtesy of making this small change that does not affect the operation.

x OpenFIREFeedback.cpp
The rumble pin is used in analog PWM mode, therefore analogWrite must be used and not digitalWrite and even though with the library for rp2040 it still works when the low level of the pin is set (with this library it is equivalent to writing analogWrite(pin, 0)), with other micros such as esp32 it does not work and you must use analogWrite setting the value to 0. Therefore I ask the courtesy, since I have adapted OpenFire to also work with esp32, to make this modification, which does not involve anything different in the operation. I have tested it both on rp2040 and on esp32 and with analogWrite it works well with both micros.

alessandro-satanassi committed:
Update SamcoPreferences.cpp

The EEPROM.update method does not exist in other EEPROM libraries such as the one for the esp32, while the EEPROM.put method exists for all libraries. The EEPROM.update method in rp2040 does nothing but call the EEPROM.put method and nothing more. I have adapted OpenFIRE to work also on EPS32 and if there are no problems, I ask the courtesy of making this small change that does not affect the operation.

alessandro-satanassi committed:
Update OpenFIREFeedback.cpp

The rumble pin is used in analog PWM mode, therefore analogWrite must be used and not digitalWrite and even though with the library for rp2040 it still works when the low level of the pin is set (with this library it is equivalent to writing analogWrite(pin, 0)), with other micros such as esp32 it does not work and you must use analogWrite setting the value to 0. Therefore I ask the courtesy, since I have adapted OpenFire to also work with esp32, to make this modification, which does not involve anything different in the operation. I have tested it both on rp2040 and on esp32 and with analogWrite it works well with both micros.

SeongGino commented on SamcoEnhanced/OpenFIREFeedback.cpp:
```diff
@@ -253,7 +253,7 @@ void OF_FFB::BurstFire()
+    analogWrite(SamcoPreferences::pins[OF_Const::rumblePin], 0);
     solenoidFirstShot = false;
     rumbleHappening = false;
     rumbleHappened = false;
```
Um, why are we touching FFB here in a PR that has nothing to do with FFB?

Besides that, the digital write OFF is meant to be unambiguous, because on RP2040 a PWM machine is being generated with any analogWrite - digitalWrites releases this, and it's necessary as the RP2040 has some limitations regarding what active analog pins can be used (so minimizing active PWMs when they aren't in use is preferred). [See core documentation.](https://arduino-pico.readthedocs.io/en/latest/analog.html#void-analogwrite-pin-size-t-pin-int-val)

alessandro-satanassi replied:
Ok, I didn't know about these 'issues' for rp2040. So I execute one instruction or the other depending on the micro used. Anyway I wanted to congratulate you for the excellent work you did with this new version of the code.

SeongGino reviewed (changes requested):
Besides a change that's out of scope, it should be noted that EEPROM usage for preferences data will be fazed out in very near future for a different solution.

alessandro-satanassi committed:
Update OpenFIREcommon.cpp

LOW is more appropriate than false, although false also works.

alessandro-satanassi committed:
Update OpenFIREserial.cpp

Put everything in a single printf, so the information is not sent to the serial in two different 'packets'. With serial via USB it creates no problems, but with wireless serial connections it occasionally does.

alessandro-satanassi committed:
Update LightgunButtons.cpp

Except for the conditions that should never occur and in any case it would probably be more logical to set the central position (default)

SeongGino commented:
Now you're touching even more parts of the code with unnecessary changes. Why are you doing all of that *in this PR?*

alessandro-satanassi commented:
Sorry for the mess I made with Git, but I still don't know how to use it well.
Sorry also for sending you so many small in my opinion cleanings/changes, by now I have studied a good part of the code, and in making it work also with esp32 every now and then I notice something and I also remove some doubts.
Congratulations for the reorganization of the code.

SeongGino commented:
Well, learn to use branches instead of just writing every commit to the main branch and sending it *while you're writing other code.*

Since the commits added are very out of scope, I'll have to close this. If you still feel obligated to commit these changes, then add them to a new branch and PR that.
